### PR TITLE
interface management improvement

### PR DESCRIPTION
**Pull request by AxelVivienCosmo**

### CODE REVIEW

File: src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java [hunk 1]
```diff
@@ -47,41 +45,46 @@ public NetworkProfile(JsonObject jsonObject) throws Exception {
     this.duplicate = jsonObject.getInt("duplicate", 0);
     this.bandwidth = jsonObject.getInt("bandwidth", 0);
     this.command = jsonObject.containsKey("command") ? jsonObject.getString("command") : this.setCommand();
-    this.nit = this.command.contains(INTERFACE_0_NAME) ? INTERFACE_0_NAME : INTERFACE_1_NAME;
-    
+    this.nit = this.command.split("dev")[1].split(" ")[1].trim();
```
Comment (ManuCosmo) [inline src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java]: this is not safe. the command can be anything

File: src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java [hunk 2]
```diff
@@ -98,7 +101,18 @@ protected void setCleanUpCommand() {
   }
 
   protected void setDefaultNit() {
-    this.nit = this.command.contains(INTERFACE_0_NAME) ? INTERFACE_0_NAME : INTERFACE_1_NAME;
+    this.nit = this.command.split("dev")[1].split(" ")[1].trim();
+    int i = 1;
```
Comment (ManuCosmo) [inline src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java]: repeated code

File: src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java [hunk 3]
```diff
@@ -47,41 +45,46 @@ public NetworkProfile(JsonObject jsonObject) throws Exception {
...
 
   protected String setCommand() throws Exception {
     String command;
+    String nit = "eth";
     String egress_command = ""; // command for traffic going out
     String ingress_command = ""; // command for traffic going in
     command =
         "sudo ip link add ifb0 type ifb || true && sudo ip link set up dev ifb0 || true && sudo tc qdisc add dev "
-            + INTERFACE_0_NAME + " ingress || true && sudo tc filter add dev " + INTERFACE_0_NAME
+            + nit + " ingress || true && sudo tc filter add dev " + nit
```
Comment (ManuCosmo) [inline src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java]: the command can be this.setCommand();, and is used to get the nit?

File: src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java [hunk 4]
```diff
@@ -32,47 +32,52 @@
 
   private final String type;
   private final String name;
-  private final String command;
-  private final String cleanUpCommand;
+  private  String command;
```
Comment (ManuCosmo) [inline src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java]: why is it necessary to remove the final?

File: src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java (hunk 1 above)
Comment (AxelVivienCosmo) [reply src/main/java/io/cosmosoftware/kite/instrumentation/NetworkProfile.java]: > we better add nit in the model as a String array

Yeah it is easier that way but who will provide nit in the model ? The client ? Us ?

**Comment by ManuCosmo** [comment]
superseded by #36

**Pull request closed without merging**